src/librustdoc/lib.rs

```diff
@@ -167,6 +167,14 @@ fn opts() -> Vec<RustcOptGroup> {
         stable("test-args", |o| {
             o.optmulti("", "test-args", "arguments to pass to the test runner", "ARGS")
         }),
+        stable("test-run-directory", |o| {
```

Comment:
I would prefer to make this unstable for now so we can see if it actually solves the issue.
```suggestion
        unstable("test-run-directory", |o| {
```